### ISSUE TITLE
-Winfinite-recursion in drivers/iio/frequency/adrf6780.c

### ISSUE THREAD

**Issue by nathanchance** [ClangBuiltLinux/linux#1490]
```
drivers/iio/frequency/adrf6780.c:117:1: error: all paths through this function will call itself [-Werror,-Winfinite-recursion]
{
^
drivers/iio/frequency/adrf6780.c:138:1: error: all paths through this function will call itself [-Werror,-Winfinite-recursion]
{
^
2 errors generated.
```

Patch submitted: https://lore.kernel.org/r/20211022195656.1513147-1-nathan@kernel.org/

**Comment by nathanchance** [comment]
Patch accepted: https://git.kernel.org/gregkh/char-misc/c/7b473ae754fec86cfa44f05f0882d1329d244e6c

**Comment by nathanchance** [comment]
Fixed in next-20211025.
